pkg/driver/provisioner_test.go

```diff
@@ -0,0 +1,406 @@
+package driver_test
+
+import (
+	"context"
+	"errors"
+	"os"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/scality/cosi/pkg/driver"
+	s3client "github.com/scality/cosi/pkg/util/s3client"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	cosiapi "sigs.k8s.io/container-object-storage-interface-spec"
+)
+
+type MockS3Client struct {
+	CreateBucketFunc func(ctx context.Context, input *s3.CreateBucketInput, opts ...func(*s3.Options)) (*s3.CreateBucketOutput, error)
+}
+
+func (m *MockS3Client) CreateBucket(ctx context.Context, input *s3.CreateBucketInput, opts ...func(*s3.Options)) (*s3.CreateBucketOutput, error) {
+	if m.CreateBucketFunc != nil {
+		return m.CreateBucketFunc(ctx, input, opts...)
+	}
+	return &s3.CreateBucketOutput{}, nil
+}
+
+var _ = Describe("ProvisionerServer DriverCreateBucket", func() {
```

Comment:
Because you're modifying the global state in the `BeforeEach`/`AfterEach` with changing `driver.InitializeClient`, I think to be safe you need to pass:

```suggestion
var _ = Describe("ProvisionerServer DriverCreateBucket", Ordered, func() {
```

To enforce running tests one by one in order.

It should be the default when running the tests without specific options, but test runs can potentially be parallelized, so this would ensure we don't modify the global state out-of-order.

Reply:
Makes sense, less falkyness in the future for other tests  as well. I will keep that in mind for future tests
fixed in https://github.com/scality/cosi/commit/414fd4b6758719043e301e250785d5218cccadb6